src/Core/Util/Number/NumberExtractor.php

```diff
@@ -64,13 +65,27 @@ public function __construct(PropertyAccessorInterface $propertyAccessor)
      *
      * ->extract($simpleArray, '[someKey]')
      *
-     * @param array|stdClass $resource
+     * @param array|object $resource
      * @param string $propertyPath
+     * @param bool $prioritizePublicProperty If true - object's public property  will be extracted first,
+     *                                       else it will search for getters.
+     *                                       Note: this will only work when providing exact property name,
+     *                                       but not path selector for inner objects
      *
      * @return Number
+     *
+     * @throws NumberExtractorException
      */
-    public function extract($resource, string $propertyPath): Number
+    public function extract($resource, string $propertyPath, bool $prioritizePublicProperty = true): Number
```

Comment:
Not sure about this additional field I would say either it's a configuration for the service (from the constructor or a setter)
Or we provide an additional method `extractFromPublicProperty` (that would be called in `extract` by default)

Reply:
removed it entirely.